### PR TITLE
feat: added searchcount component

### PR DESCRIPTION
**Pull request by BlakeJC94**
Hello :wave: Now with `cmdheight=0`, the search count `[x/x]` that used to be in the right of the command mode buffer when navigating search results is hidden. I added a simple function to restore it via `lualine`, and thought it might be worthwhile to contribute upstream.

Happy to add a bit of documentation in the appropriate locations if this is a welcomed idea

### CODE REVIEW

**Comment by shadmansaleh** [comment]
With cmdheight 0 option now I think it makes sense to add it. You'll have to add it to the component list in README.md for this pr to be merged.

Also you can take a look at https://github.com/nvim-lualine/lualine.nvim/issues/186 we previously had discussed it in there.

**Comment by BlakeJC94** [comment]
Absolutely, I'll have a look at adding this to the docs when I get a bit of time today

**Comment by BlakeJC94** [comment]
@shadmansaleh ready to go now, added docs and improved the code after reading the linked issue. Let me know if there's anything else I can do to help :+1:

File: README.md [hunk 1]
```diff
@@ -249,6 +249,7 @@ sections = {lualine_a = {'mode'}}
 - `filetype`
 - `hostname`
 - `location` (location in file in line:column format)
+- `searchcount` (number of search matches when hlsearch is active)
```
Comment (shadmansaleh) [inline README.md:252]: the list is alphabetically sorted :]

Comment (BlakeJC94) [reply README.md]: Oh my! Apologies for that, tried to get this done quickly during a commute. Fixed in the latest commit :)

**Review by shadmansaleh — APPROVED** [review]

**Pull request merged**